packages/lib-utils/src/k8s/hooks/useK8sWatchResource.test.ts

```diff
@@ -0,0 +1,179 @@
+import { renderHook } from '@testing-library/react-hooks/native';
+import { Map as ImmutableMap } from 'immutable';
+import { useSelector } from 'react-redux';
+import type { K8sModelCommon, K8sResourceCommon } from '../../types/k8s';
+import type { WatchData } from './k8s-watch-types';
+import type { NoModelError } from './k8s-watcher';
+import { getWatchData, getReduxData } from './k8s-watcher';
+import { useDeepCompareMemoize } from './useDeepCompareMemoize';
+import { useK8sModel } from './useK8sModel';
+import { useK8sWatchResource } from './useK8sWatchResource';
+import { useModelsLoaded } from './useModelsLoaded';
+import type { WatchK8sResource } from './watch-resource-types';
+
+const watchedResourceMock: WatchK8sResource = {
+  isList: false,
+  groupVersionKind: {
+    group: 'appstudio.redhat.com',
+    version: 'v1alpha1',
+    kind: 'Application',
+  },
+  name: 'test',
+  namespace: 'test-ns',
+};
+
+const resourceModelMock: K8sModelCommon | undefined = {
+  apiGroup: 'appstudio.redhat.com',
+  apiVersion: 'v1alpha1',
+  crd: true,
+  kind: 'Application',
+  namespaced: true,
+  plural: 'applications',
+};
+
+const resourceDataMock: K8sResourceCommon = {
+  apiVersion: 'appstudio.redhat.com/v1alpha1',
+  kind: 'Application',
+  metadata: {
+    annotations: { finalizeCount: '0', test: 'patched-value' },
+    creationTimestamp: '2022-04-29T13:41:21Z',
+    finalizers: ['application.appstudio.redhat.com/finalizer'],
+    generation: 1,
+    name: 'test',
+    namespace: 'vnambiar',
+    resourceVersion: '414309692',
+    uid: '602ad43f-1a71-4e71-9314-d93bffbc0762',
+  },
+  spec: { appModelRepository: { url: '' }, displayName: '', gitOpsRepository: { url: '' } },
+  status: {
+    conditions: [
+      {
+        lastTransitionTime: '2022-04-29T13:41:22Z',
+        message: 'Application has been successfully created',
+        reason: 'OK',
+        status: 'True',
+        type: 'Created',
+      },
+    ],
+    devfile:
+      'metadata:\\n  attributes:\\n    appModelRepository.context: /\\n    appModelRepository.url: https://github.com/redhat-appstudio-appdata/-vnambiar-touch-drive\\n    gitOpsRepository.context: /\\n    gitOpsRepository.url: https://github.com/redhat-appstudio-appdata/-vnambiar-touch-drive\\nschemaVersion: 2.1.0\\n',
+  },
+};
+
+let undefinedModelMock: K8sModelCommon | undefined;
+
+// Mock modules
+
+jest.mock('./k8s-watcher', () => ({
```

Comment:
@vidyanambiar I'll need to look at our Jest config, but we should ensure these mocks are cleared/reset between test files to prevent cross-contamination.

Reply:
@florkbr We do have `clearMocks` set to `true` in the [jest config](https://github.com/openshift/dynamic-plugin-sdk/blob/main/packages/common/test-configs/jest-config-base.ts#L15) at the moment. I recall @vojtechszocs had recommended it's better to explicitly clear it in the test code. 
I have `jest.clearAllMocks` in the [afterEach](https://github.com/openshift/dynamic-plugin-sdk/pull/85/files#diff-37b388a5fd51bb4dc2241412abafe4a68c75dc9bc1fe30389fd40d7c5ab1ceb7R94) hook.  @vojtechszocs @florkbr What's your recommendation? Shall I remove the global `clearMocks: true` from the jest.config or get rid of the afterEach hook where I'm clearing mocks?

Reply:
@vidyanambiar I think we can revisit this, however my points would be:
1. `clearMocks` set to `true` in the jest config is good - however that won't remove the mock implementations themselves. That's what `resetMocks` is for.
2. It's not clear from the doc if the global mocks will be removed (IE the mocks on react/etc). We should be able to test that - IE add another test file without mocking the global modules we have in these 2 files and see if the test passes.

I personally think keeping clearMocks is a good thing (as a fallback if we fail to clean up some mocks). However I agree with @vojtechszocs that we should also clean up after ourselves in the tests.

Reply:
@florkbr As you suggested, I added another test file without mocking the global modules we have in these 2 test files to see if the test passes. The test does not pass and jest appears to be using the unmocked version of the module as expected. I found that this still happens even if we comment out clearAllMocks in the previous files as well as in the jest config.
So it appears that clearAllMocks applies within the context of a test file, if we need to clear the mock between individual tests in a file rather than between test files. I found some more info here. https://stackoverflow.com/questions/44141276/does-jest-automatically-restore-mocked-modules-between-test-modules#:~:text=You%20don't%20have%20to,in%20its%20own%20sandboxed%20thread. Let me know what you think.
PS: The only global mock that we seem to have right now is `react` which was added as part of [setup-react.ts](https://github.com/openshift/dynamic-plugin-sdk/blob/main/packages/common/test-configs/setup-react.ts).